pypesto/visualize/parameters.py

```diff
@@ -381,14 +381,13 @@ def handle_inputs(
         res.get(INNER_PARAMETERS, None) for res in result.optimize_result.list
     ]
     if any(inner_xs):
-        # search for first non-empty inner_xs to obtain inner_xs_names
-        for inner_xs_idx in inner_xs:
-            if inner_xs_idx is not None:
-                inner_xs_names = list(inner_xs_idx.keys())
-                break
-        # fill inner_xs with nan if no inner_xs are available
+        inner_xs_names = next(
+            list(inner_xs_idx.keys())
+            for inner_xs_idx in inner_xs
+            if inner_xs_idx is not None
+        )
```

Comment:
In the problem there is a dictionary `problem.objective.calculator.inner_problem.xs` with keys as names of the parameters.
But yeah, I made it like this so it doesn't depend on the problem in the result. 

On another note, it might be nice to have these names in the `problem` directly instead of far deep in the `inner_problem`...